MyNumSetLibrary/MyNumSetLibrary.Tests/NumSetTests.cs

```diff
@@ -0,0 +1,239 @@
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace MyNumSetLibrary.Tests
+{
+    [TestClass]
+    public class NumSetTests
+    {
+        [TestMethod]
+        [ExpectedException(typeof(ArgumentNullException))]
+        public void NumSet_Constructor_ThrowsArgumentNullException()
+        {
+            //Assign
+            _ = new NumSet(null!);
+
+            //Act
+
+            //Assert
+        }
+
+        [TestMethod]
+        [ExpectedException(typeof(ArgumentNullException))]
+        public void NumSet_ToString_ThrowsArgumentNullException()
+        {
+            //Assign
+            NumSet numSet = new(null!);
+
+            //Act
+            numSet.ToString();
+
+            //Assert
+        }
+
+        [TestMethod]
+        [DataRow(new int[] { }, "")]
+        [DataRow(new int[] { 1, 2, 3, 4, 5 }, "1,2,3,4,5")]
+        public void NumSet_ToString(int[] array, string expected)
+        {
+            //Assign
+            NumSet numSet = new(array!);
+
+            //Act
+
+            //Assert
+            Assert.AreEqual(expected, numSet.ToString());
+        }
+
+        [TestMethod]
+        public void NumSet_Equals_FalseWithNull()
+        {
+            //Assign
+            int[] array = Array.Empty<int>();
+            NumSet left = new(array);
+
+            //Act
+
+            //Assert
+            Assert.IsFalse(left.Equals(null));
```

Comment:
nit: Left is a bit meaningless here
```suggestion
            NumSet set = new(array);

            //Act

            //Assert
            Assert.IsFalse(set.Equals(null));
```